plugins/text-editor/src/types.ts

```diff
@@ -145,3 +147,50 @@ export interface TextEditorExtensionFactory extends Doc {
   index: number
   create: Resource<ExtensionCreator>
 }
+
+/**
+ * Action handler for text editor action
+ */
+export type TextActionFunction = (editor: Editor, event: MouseEvent, ctx: ActionContext) => Promise<void>
+
+/**
+ * Handler to determine whether the text action is visible
+ */
+export type TextActionVisibleFunction = (editor: Editor, ctx: ActionContext) => Promise<boolean>
+
+/**
+ * Handler to determine whether the text action is active
+ */
+export type TextActionActiveFunction = (editor: Editor) => Promise<boolean>
+
+/**
+ * Describes toggle handler for a text action
+ */
+export interface TogglerDescriptor {
+  command: string
```

Comment:
let's try to make it more type safe to ensure that editor commands exists, at least when we declare this action

Reply:
Done